modules/solid_mechanics/include/materials/ReturnMappingModel.h

```diff
@@ -8,61 +8,79 @@
+                             SymmTensor & stress_new) override;
 
-  void computeStress(const Elem & /*current_elem*/,
-                     unsigned qp,
+  /**
+   * Compute stress by performing return mapping iterations.  This can be called either
+   * from within this model, or by an external model that combines multiple inelastic models.
+   * @param current_elem               Current element
+   * @param elasticityTensor           Elasticity tensor
+   * @param stress_old                 Old state of stress
+   * @param strain_increment           Strain increment
+   * @param stress_new                 New state of stress
+   * @param inelastic_strain_increment Inelastic strain increment
+   */
+  void computeStress(const Elem & current_elem,
                      const SymmElasticityTensor & elasticityTensor,
                      const SymmTensor & stress_old,
                      SymmTensor & strain_increment,
                      SymmTensor & stress_new,
                      SymmTensor & inelastic_strain_increment);
 
-  Real computeTimeStepLimit(unsigned qp);
+  virtual Real computeReferenceResidual(const Real effective_trial_stress,
+                                        const Real scalar) override;
+
+  virtual Real getDT() override { return _dt; }
+
+  /**
+   * Compute the limiting value of the time step for this material
+   * @return Limiting time step
+   */
+  Real computeTimeStepLimit();
 
 protected:
-  virtual void computeStressInitialize(unsigned /*qp*/,
-                                       Real /*effectiveTrialStress*/,
+  /**
+   * Perform any necessary initialization before return mapping iterations
+   * @param effectiveTrialStress Effective trail stress
+   * @param elasticityTensor     Elasticity tensor
+   */
+  virtual void computeStressInitialize(Real /*effectiveTrialStress*/,
                                        const SymmElasticityTensor & /*elasticityTensor*/)
   {
   }
-  virtual void computeStressFinalize(unsigned /*qp*/,
-                                     const SymmTensor & /*inelasticStrainIncrement*/)
-  {
-  }
 
-  virtual void iterationInitialize(unsigned /*qp*/, Real /*scalar*/) {}
-  virtual Real computeResidual(unsigned qp, Real effectiveTrialStress, Real scalar) = 0;
-  virtual Real computeDerivative(unsigned qp, Real effectiveTrialStress, Real scalar) = 0;
-  virtual void iterationFinalize(unsigned /*qp*/, Real /*scalar*/) {}
+  /**
+   * Perform any necessary steps to finalize state after return mapping iterations
+   * @param inelasticStrainIncrement Inelastic strain increment
+   */
+  virtual void computeStressFinalize(const SymmTensor & /*inelasticStrainIncrement*/) {}
 
-  const unsigned int _max_its;
-  const bool _output_iteration_info;
-  const bool _output_iteration_info_on_error;
-  const Real _relative_tolerance;
-  const Real _absolute_tolerance;
   Real _effective_strain_increment;
 
+  /// 3 * shear modulus
+  Real _three_g;
```

Comment:
Sure, I can make that change.  I think the standard is that we're supposed to Doxygen coment all of these variables, even if they have self-explanatory names, which gives more opportunities for comments that are lies.  Did I mention that I see no value in Doxygen?